crates/proof-of-sql/src/proof_primitive/dory/extended_dory_reduce_helper.rs

```diff
@@ -104,12 +104,19 @@ pub fn extended_dory_reduce_verify_update_Es(
 /// `s1_tensor[nu-1] <- s1_tensor[nu-1] * (1- alpha) + alpha`
 ///
 /// and taking the product in [fold_scalars_0_verify](super::fold_scalars_0_verify).
-pub fn extended_dory_reduce_verify_fold_s_vecs(
-    state: &mut ExtendedVerifierState,
-    (alpha, alpha_inv): (F, F),
-) {
-    state.s1_tensor[state.base_state.nu - 1] *= F::ONE - alpha;
-    state.s1_tensor[state.base_state.nu - 1] += alpha;
-    state.s2_tensor[state.base_state.nu - 1] *= F::ONE - alpha_inv;
-    state.s2_tensor[state.base_state.nu - 1] += alpha_inv;
+pub fn extended_dory_reduce_verify_fold_s_vecs(state: &ExtendedVerifierState) -> (F, F) {
+    (
+        state
+            .s1_tensor
+            .iter()
+            .zip(state.alphas.iter())
+            .map(|(s, a)| (F::ONE - s) * a + s)
+            .product(),
+        state
+            .s2_tensor
+            .iter()
+            .zip(state.alpha_invs.iter())
+            .map(|(s, a)| (F::ONE - s) * a + s)
+            .product(),
```

Comment:
Is `s` an element of the tensor and `a` an element of alpha? I would think `.map(|(s, a)| (F::ONE - s) * a + s)`  should be `.map(|(s, a)| (F::ONE - a) * s + a)` to match the previous code? But the tests are passing, so I might be misunderstanding something?

Reply:
Good eye. This is why it works:

$$(1 - s) \cdot a + s = a - s\cdot a + s = s - a\cdot s + a = (1 - a) \cdot s + a$$